src/Lecture2.hs

```diff
@@ -74,7 +79,14 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+removeAt :: Int -> [a] -> (Maybe a, [a])
+removeAt ix xs
+  | null xs = (Nothing, [])
+  | ix < 0 || ix >= length xs = (Nothing, xs)
```

Comment:
`length` hangs on infinite lists that's why `removeAt` also doesn't work on infinite lists ♾️